lesson7/task2/handlers/handle.go

```diff
@@ -0,0 +1,102 @@
+package handlers
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+	"log"
+	"net/http"
+
+	"github.com/xruterx/golang/lesson7/task2/handlers/methods"
+)
+
+type Request struct {
+	Method string          `json:"method"`
+	Params json.RawMessage `json:"params"`
+	Id     int64           `json:"id"`
+}
+
+type Response struct {
+	Id     int64       `json:"id"`
+	Error  interface{} `json:"error"`
+	Result interface{} `json:"result"`
+}
+
+func Handle(w http.ResponseWriter, r *http.Request) {
+	req, err := parseRequest(r)
+	if err != nil {
+		returnErr(w, -1, err.Error())
+		return
+	}
+	resp, err := handle(*req)
+	if err != nil {
+		returnErr(w, req.Id, err.Error())
+		return
+	}
+	returnOK(w, req.Id, resp)
+}
+
+func parseRequest(r *http.Request) (*Request, error) {
+	body := r.Body
+	defer body.Close()
+	bodyData, err := io.ReadAll(body)
+	if err != nil {
+		log.Print(err.Error())
+		return nil, errors.New("failed to read request body")
```

Comment:
It is good GO style to have a block of constants in the top of the file with all errors messages